test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js

```diff
@@ -3,865 +3,267 @@
+        expect(putLogStatement.Resource).to.deep.includes({
+          'Fn::Sub':
+            'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/myCustomName:*:*',
+        });
       });
 
       it('should configure LogGroup resources for functions', () => {
-        // Replaces
-        // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L546-L585
+        const myFunctionWithRole = naming.getLogGroupLogicalId('myFunctionWithRole');
+        const myCustomName = cfResources[myFunctionWithRole];
+
+        expect(myCustomName.Type).to.be.equal('AWS::Logs::LogGroup');
+        expect(myCustomName.Properties.LogGroupName).to.be.equal('/aws/lambda/myCustomName');
+
+        const myFunctionName = naming.getLogGroupLogicalId('myFunction');
+        const myFunctionResource = cfResources[myFunctionName];
+
+        expect(myFunctionResource.Type).to.be.equal('AWS::Logs::LogGroup');
+        expect(myFunctionResource.Properties.LogGroupName).to.be.equal(
+          '/aws/lambda/another-service-dev-myFunction'
+        );
       });
     });
 
-    describe.skip('TODO: Provider properties', () => {
+    describe('Provider properties', () => {
+      let cfResources;
+      let naming;
+
       before(async () => {
         await runServerless({
           fixture: 'function',
-          configExt: {},
+          cliArgs: ['package'],
+          configExt: {
+            service: 'my-service',
+            provider: {
+              vpc: {
+                securityGroupIds: ['xxx'],
+                subnetIds: ['xxx'],
+              },
+              logRetentionInDays: 5,
+              iamManagedPolicies: [
+                'arn:aws:iam::123456789012:user/*',
+                'arn:aws:s3:::my_corporate_bucket/Development/*',
+                'arn:aws:iam::123456789012:u*',
+              ],
+              rolePermissionsBoundary: ['arn:aws:iam::123456789012:policy/XCompanyBoundaries'],
+            },
+          },
+        }).then(({ cfTemplate, awsNaming }) => {
+          cfResources = cfTemplate.Resources;
+          naming = awsNaming;
         });
       });
 
-      it('should support `provider.iamRoleStatements`', () => {
-        // Replaces
-        // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L412-L428
-        // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L492-L508
-        // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L510-L526
-      });
+      it('should support `provider.iamRoleStatements`', async () => {
+        const {
+          cfTemplate: { Resources },
+          awsNaming,
+        } = await runServerless({
```

Comment:
There should be no `runServerless` run here. It should be tested through service setup in `before` clause